turbopack/crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -2835,6 +2843,16 @@ fn emit_unresolveable_issue(
     .emit();
 }
 
+async fn emit_issue(mut issue: ResolvingIssue) -> Result<()> {
+    if !(*issue.severity.await? == IssueSeverity::Error
+        && issue.resolve_options.await?.loose_errors)
+    {
+        issue.severity = IssueSeverity::Warning.cell();
+    }
```

Comment:
I don't like that this is reading the just `cell`ed value again. A lot of unnecessary logic where it would be more efficient to construct ResolvingIssue with the right severity directly.

Maybe add a helper `error_severity(ResolveOptions)` since `emit_issue` is only called with `IssueSeverity::Error` anyway.